src/codeu/chat/util/PlayInfo.java

```diff
@@ -28,26 +33,48 @@ public void write(OutputStream out, PlayInfo value) throws IOException {
     public PlayInfo read(InputStream in) throws IOException {
       String title = Serializers.STRING.read(in);
       String status = Serializers.STRING.read(in);
-      return new PlayInfo(title, status);
+      return new PlayInfo(title, status, false);
     }
   };
 
-  public PlayInfo(String title, List<String> openRoles) {
-    roles = new HashMap<>();
+  public PlayInfo(String title, String playTitle) {
     this.title = title;
-    this.openRoles = openRoles;
+
+    roles = new HashMap<>();
+    openRoles = new ArrayList<>();
+    current_part = 1;
+    this.parts = 3;
     lines = new ArrayDeque<>();
-    load50Lines();
+    load50Lines(playTitle, current_part);
+    loadRoles(playTitle);
   }
 
-  public PlayInfo(String title, String status) {
+  public PlayInfo(String title, String status, boolean b) {
     this.title = title;
     this.status = status;
   }
 
-  public void setRole(Uuid user) {
-    String character = openRoles.remove(0);
-    roles.put(character, user);
+  public void setConversation(ConversationHeader c) {
```

Comment:
Although here it doesn't affect code readability, try to get into the habit of using descriptive variable names.